src/forms/DepositForm.vue

```diff
@@ -203,33 +212,31 @@ const fetchAllowanceByCurrency = async (
 ): Promise<BigNumber> => {
   let contract
   switch (currency) {
-    case CURRENCIES.stEth:
-      contract = web3ProvidersStore.stEthContract
+    case CURRENCIES.depositToken:
+      contract = web3ProvidersStore.depositContract
       break
     default:
       throw new Error('unknown currency')
   }
 
   return contract.providerBased.value.allowance(
     web3ProvidersStore.provider.selectedAddress,
-    config.networksMap[web3ProvidersStore.networkId].contractAddressesMap
-      .erc1967Proxy,
+    web3ProvidersStore.erc1967ProxyContract.providerBased.value.address,
   )
 }
 
 const approveByCurrency = async (currency: CURRENCIES) => {
   let contract
   switch (currency) {
-    case CURRENCIES.stEth:
-      contract = web3ProvidersStore.stEthContract
+    case CURRENCIES.depositToken:
+      contract = web3ProvidersStore.depositContract
       break
     default:
       throw new Error('unknown currency')
```

Comment:
Since you have `ErrorHandler` singleton that processes all app errors based on error custom class - there shouldn't be a place in app where you just throw plain `Error`.

Create error class for this error + add new error handling to `ErrorHandler`

P.S. check also other places where you could've thrown unspecified errors

Reply:
done